src/api/app/models/user.rb

```diff
@@ -661,6 +666,10 @@ def delete!
       project.destroy
     end
 
+    # wipe affected caches
+    myself.involved_projects.each do |prj| prj.reset_cache; end
+    myself.involved_packages.each do |pkg| pkg.reset_cache; end
```

Comment:
This does nothing to resolve your problem. There will still be a `Relationship` between User and Project...